mock/src/transaction.rs

```diff
@@ -1,51 +1,155 @@
 //! Mock Transaction definition and builder related methods.
 
 use super::{MOCK_ACCOUNTS, MOCK_CHAIN_ID, MOCK_GASPRICE};
-use eth_types::{AccessList, Address, Bytes, Hash, Transaction, Word, U64};
+use eth_types::word;
+use eth_types::{
+    geth_types::Transaction as GethTransaction, AccessList, Address, Bytes, Hash, Transaction,
+    Word, U64,
+};
+use ethers_core::{
+    rand::{CryptoRng, RngCore},
+    types::TransactionRequest,
+    utils::keccak256,
+};
+use ethers_signers::{LocalWallet, Signer};
+use lazy_static::lazy_static;
+use rand::SeedableRng;
+use rand_chacha::ChaCha20Rng;
+
+lazy_static! {
+    /// Collection of correctly hashed and signed Transactions which can be used to test circuits or opcodes that have to check integrity of the Tx itself.
+    /// Some of the parameters of the Tx are hardcoded such as `nonce`, `value`, `gas_price` etc...
+    pub static ref CORRECT_MOCK_TXS: Vec<MockTransaction> = {
+        let mut rng = ChaCha20Rng::seed_from_u64(2u64);
+
+        vec![MockTransaction::default()
+            .from(AddrOrWallet::random(&mut rng))
+            .to(MOCK_ACCOUNTS[0])
+            .nonce(word!("0x103"))
+            .value(word!("0x3e8"))
+            .gas_price(word!("0x4d2"))
+            .input(Bytes::from(b"hello"))
+            .build(),
+            MockTransaction::default()
+            .from(AddrOrWallet::random(&mut rng))
+            .to(MOCK_ACCOUNTS[1])
+            .nonce(word!("0x104"))
+            .value(word!("0x3e8"))
+            .gas_price(word!("0x4d2"))
+            .input(Bytes::from(b"hello"))
+            .build(),
+            MockTransaction::default()
+            .from(AddrOrWallet::random(&mut rng))
+            .to(MOCK_ACCOUNTS[2])
+            .nonce(word!("0x105"))
+            .value(word!("0x3e8"))
+            .gas_price(word!("0x4d2"))
+            .input(Bytes::from(b"hello"))
+            .build()]
+    };
+}
+
+#[derive(Debug, Clone)]
+pub enum AddrOrWallet {
+    Addr(Address),
+    Wallet(LocalWallet),
+}
+
+impl Default for AddrOrWallet {
+    fn default() -> Self {
+        AddrOrWallet::Addr(Address::default())
+    }
+}
+
+impl From<Address> for AddrOrWallet {
+    fn from(addr: Address) -> Self {
+        AddrOrWallet::Addr(addr)
+    }
+}
+
+impl From<LocalWallet> for AddrOrWallet {
+    fn from(wallet: LocalWallet) -> Self {
+        AddrOrWallet::Wallet(wallet)
+    }
+}
+
+impl AddrOrWallet {
+    /// Generates a random Wallet from a random secpk256 keypair
+    pub fn random<R: RngCore + CryptoRng>(rng: &mut R) -> Self {
```

Comment:
(no changes proposed in this comment)
idea: we can define a `Randomizable` trait so that we can turn structs into random instances by calling `Foo::random(rng)`

Reply:
The general way on which all of the libs do the `random` is on that way. There's no such trait that does what you mean.
I don't think it's a bad idea. But define a trait just for this function for this structure seems odd. I think the only place where a `random` trait makes sense is in `rand` or `rand_core` libs.

Anyway, as said, I like your vision on that!

Reply:
Randomizable can be applied to blocks, txs, or all other factor objects.